apis/duck/v1alpha1/sinkable_types.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Sinkable is very similar concept as Targetable. However, at the
+// transport level they have different contracts and hence Sinkable
+// and Targetable are two distinct resources.
+// It would be better to have one level of indirection from Status.
+// The way this is currently put in at the same level as the other Status
+// resources. Hence the objects supporting Sinkable (Knative channel for
+// example), will expose it as: Status.DomainInternal
+// For new resources that are built from scratch, they should probably
+// introduce an extra level of indirection.
+package v1alpha1
+
+import (
+	"encoding/json"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"github.com/knative/pkg/apis/duck"
+)
+
+// Sinkable is the schema for the sinkable portion of the payload
+type Sinkable struct {
+	DomainInternal string `json:"domainInternal,omitempty"`
```

Comment:
How is Sinkable different from Targetable?

Reply:
We have an idea that they are different in the data-plane. But they look the same in the control plane.

Reply:
Look at Line 17-25 above :) If it's not clear, lmk and I'll update. IIRC, I also moved it down here if it's easier to find, mayhaps I put it into a nonstd location above the package.

Reply:
Also replicated the comment to targetable for extra visiblity.